src/gradletestrunnersmoketest.sh

```diff
@@ -18,14 +18,24 @@ gradletestrunnersmoketest () {
   ./gradlew spotlessCheck
   ./gradlew spotbugsMain
 
+  outputDir="tmp/TestRunnerResults"
+
+  # if runTest fails, then try to uploadResults before exiting with an error
   echo "Running test suite"
-  ./gradlew runTest --args="suites/PRSmokeTests.json tmp/TestRunnerResults"
+  ./gradlew runTest --args="suites/PRSmokeTests.json $outputDir" ||
+    echo "Running test suite failed, Uploading results" &&
```

Comment:
If I'm reading this correctly...wouldn't this always print "Running test suite failed?"

Reply:
No, it should only execute the commands after the || if the first command returns false.

```
marikomedlock-macbookpro:datarepo-clienttests marikomedlock$ true || echo "after OR"
marikomedlock-macbookpro:datarepo-clienttests marikomedlock$ false || echo "after OR"
after OR
```

Reply:
Ah got it...my bash-fu is weak